### ISSUE TITLE
Bug: Timeline chart throws error and does not render after update

### ISSUE THREAD

**Issue by orwant** [google/google-visualization-issues#1641]
```
What steps will reproduce the problem? Please provide a link to a
demonstration page if at all possible, or attach code.
1. Load page with Timeline graph.
2. Timeline doesn't display, an error is shown instead.

What component is this issue related to (PieChart, LineChart, DataTable,
Query, etc)?
Timeline


Are you using the test environment (version 1.1)?
(If you are not sure, answer NO)
No

What operating system and browser are you using?
Chrome 35, Windows 7

We have narrowed the problem down to two parts:
1. Timeline no longer handles ‘nulls’ in the ‘start’ or ‘end’ fields of the datatable
rows. 
2. Timeline ‘chart.getSelection()’ no longer returns an object with any values (previously
contained at least fromdate and todate). Only the row number is contained.

Test cases:
Part 1: see attached file. Chart does not render.
Part 2: see following snippet of code. "selecteditem" does not contain the values any
more, while it did before the update.
google.visualization.events.addListener(chart, 'select', function () {
var selectedItem = chart.getSelection()[0];
    if (selectedItem) {               
        ...
        var fromdate = dataTable.getValue(selectedItem.Fb, 2);
        var todate = dataTable.getValue(selectedItem.Fb, 3);
        ...

Thanks for your attention.

*********************************************************
For developers viewing this issue: please click the 'star' icon to be
notified of future changes, and to let us know how many of you are
interested in seeing it resolved.
*********************************************************

```

Original issue reported on code.google.com by `james2048` on 2014-06-25 16:26:41

<hr>
- _Attachment: [broken case nulls.htm](https://storage.googleapis.com/google-code-attachments/google-visualization-api-issues/issue-1641/comment-0/broken case nulls.htm)_

**Comment by orwant** [comment]
```
This issue has broken critical business logic in my companies code. We have discovered
that the .Fb identifier has changed to be .bb if this helps anyone else. But we wourld
really like to know why?
```

Original issue reported on code.google.com by `adam.garner.eposnow` on 2014-06-25 16:50:38

**Comment by orwant** [comment]
```
This really is crucial with my companies code. Needs resolving asap.
```

Original issue reported on code.google.com by `edward.layton.eposnow` on 2014-06-26 07:59:39

**Comment by orwant** [comment]
```
Disappointed by this. It's causing me no end of aggravation. Such terrible notification
about these things. I just don't think Google realises how important their platform
being stable is for the customers using it.
```

Original issue reported on code.google.com by `b@brc.co` on 2014-06-26 08:01:42

**Comment by orwant** [comment]
```
Hi guys,

This looks like two different bug reports to me: 1) nulls no longer handled, and 2)
the .Fb identifier has changed names. I'm confused about some of the details, though,
so let's see if we can clarify.

#2: As far as I can tell looking at the code, the selection object (returned by chart.getSelection())
never contained the start and end dates. It returns an array of objects with the 'row'
and 'column' fields set appropriately. (For timeline, col will always be null.) The
behavior change that we did make in the last release was to fix a bug in this object.
Our Javascript compiler was renaming 'row' and 'column' to something else. (Probably
the .Fb identifier that you reference.) The 'Fb' field is now correctly named 'row'
as described in the documentation. (https://developers.google.com/chart/interactive/docs/reference#visgetselection)

#1: Every row in the input data is required to have both a start and an end time. This
is documented. (https://developers.google.com/chart/interactive/docs/gallery/timeline#Data_Format)
Any behavior using those null values is undefined, and you shouldn't rely on it. (What
does it even mean for a task to have no start or end times?) How could we ever fix
any bug if we had to continue supporting the old broken behavior? 

In short, both of these changes are bug fixes. For the field name that has changed,
it is fixed now, and it won't change again. For the null values issue, it was undefined
behavior, and therefore we can't support it. 


```

Original issue reported on code.google.com by `gmadrid@google.com` on 2014-06-27 18:56:55

**Comment by orwant** [comment]
```
The null null start and end times showed a row with no bars in it, this is super handy,
as we need to be able to show a list which may have 30 rows but only 2 or 3 with bars
in them.
```

Original issue reported on code.google.com by `adam.garner.eposnow` on 2014-06-30 08:03:09

**Comment by orwant** [comment]
```
so if i understand this correctly now instead of

var fromdate = dataTable.getValue(selectedItem.Fb, 2);

we should do

var fromdate = dataTable.getValue(selectedItem.row, 2);

right?
this does not actually work right now. what does work is

var fromdate = dataTable.getValue(selectedItem.bb, 2);

like Adam said in comment 1.
```

Original issue reported on code.google.com by `Yossi.Rafelson` on 2014-07-03 23:56:25

**Comment by orwant** [comment]
```
The selectedItem.row functionality was finally deployed sometime in the last few days.
My code that was using selectedItem.bb broke and the fix was to use selectedItem.row

Hopefully this is the last time.
```

Original issue reported on code.google.com by `Yossi.Rafelson` on 2014-11-10 23:37:20
